src/Hazelcast.Net/ClientState.cs

```diff
@@ -72,6 +72,13 @@ public enum ClientState
         /// </remarks>
         ShuttingDown,
 
+        /// <summary>
+        /// The client switched the cluster options, and will reconnect with new options
+        /// <para>This state is reached when cluster switching is completed, 
+        /// and client will reconnect with new options</para>
```

Comment:
It is different because we have one `ClientState` parameter and event across the client for both public and internal. The event is subscribed by internal parts which should do some preparation before connecting to ne cluster.(Clearing cache, connections etc.) If we trigger the event after connection is established, then we need an internal states and event in the current structure.

Reply:
we need to change this - and use a different internal event if needed - we need to align with java - the "changed cluster" event triggers only once and only after we have successfully connected to a new cluster - otherwise... ppl may get two or three "changed cluster" but only the last one is for real.